tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+        ]
+        #yapf: enable
+        config_report = ConfigCommandReport()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='report', help='', config=config_report)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class OptionStruct():
+
+    def __init__(self,
+                 type,
+                 stage,
+                 long_flag,
+                 short_flag=None,
+                 expected_value=None,
+                 default_value=None,
+                 expected_failing_value=None,
+                 extra_commands=None):
+
+        self.long_flag = long_flag
+        self.short_flag = short_flag
+        self.expected_value = expected_value
+        self.default_value = default_value
+        self.expected_failing_value = expected_failing_value
+        self.type = type
+        self.extra_commands = extra_commands
+
+        if stage == "profile":
+            self.cli_subcommand = CLIConfigProfileStruct
+        elif stage == "analyze":
+            self.cli_subcommand = CLIConfigAnalyzeStruct
+        elif stage == "report":
+            self.cli_subcommand = CLIConfigReportStruct
+
+
+@patch('model_analyzer.config.input.config_command_profile.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch(
+    'model_analyzer.config.input.config_command_profile.binary_path_validator',
+    lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch('model_analyzer.config.input.config_command_analyze.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+@patch('model_analyzer.config.input.config_command_report.file_path_validator',
+       lambda _: ConfigStatus(status=CONFIG_PARSER_SUCCESS))
+class TestCLIOptions(trc.TestResultCollector):
+    """
+    Tests the methods of the CLI class
+    """
+
+    def test_basic_cli_config_profile_options(self):
+        """
+        Test the minimal set of cli commands necessary to run Model Analyzer profile
+        """
+        cli = CLIConfigProfileStruct()
+        _, config = cli.parse()
+        model_repo = config.model_repository
+        profile_model = config.profile_models[0].model_name()
+        self.assertEqual('foo', model_repo)
+        self.assertEqual('bar', profile_model)
+
+    @patch(
+        'model_analyzer.config.input.config_command_report.ConfigCommandReport._load_config_file'
+    )
+    @patch(
+        'model_analyzer.config.input.config_command_report.ConfigCommandReport._preprocess_and_verify_arguments'
+    )
+    @patch(
+        'model_analyzer.config.input.config_command_analyze.ConfigCommandAnalyze._load_config_file'
+    )
+    @patch(
+        'model_analyzer.config.input.config_command_analyze.ConfigCommandAnalyze._preprocess_and_verify_arguments'
+    )
+    @patch(
+        'model_analyzer.config.input.config_command_profile.ConfigCommandProfile._load_config_file'
+    )
+    def test_all_options(self, mocked_load_config_file_profile,
```

Comment:
We need a descriptive comment of all the magic, either here or at the top of the class.  People need to be able to come to this file and add new options without needing to ask you how it all works.

Reply:
I tried to add comments about each of the forms of the OptionStruct class that I used and I grouped the different types together. 
If you do not think those comments are not sufficient, lets chat really fast offline about it. I am probably too close to this to know what to describe at this point.